src/test/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshRequestsTest.java

```diff
@@ -0,0 +1,97 @@
+package uk.nhs.digital.nhsconnect.lab.results.mesh;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.Header;
+import org.apache.http.client.methods.HttpEntityEnclosingRequestBase;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.junit.jupiter.api.Test;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.http.MeshConfig;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.http.MeshHeaders;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.http.MeshRequests;
+import uk.nhs.digital.nhsconnect.lab.results.mesh.message.WorkflowId;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MeshRequestsTest {
+
+    private final MeshConfig meshConfig = new MeshConfig("mailboxId",
+        "password",
+        "SharedKey",
+        "https://localhost:8829/messageexchange/",
+        "false",
+        StringUtils.EMPTY,
+        StringUtils.EMPTY,
+        StringUtils.EMPTY);
+    private final MeshHeaders meshHeaders = new MeshHeaders(meshConfig);
+
+    @Test
+    void when_GettingMessage_Then_ExpectHttpGetAndCorrectUri() {
+        MeshRequests meshRequests = new MeshRequests(meshConfig, meshHeaders);
+
+        var request = meshRequests.getMessage("messageId");
+
+        assertThat(request).isExactlyInstanceOf(HttpGet.class);
+        assertThat(request.getURI().toString()).isEqualTo("https://localhost:8829/messageexchange/mailboxId/inbox/messageId");
```

Comment:
I find it helpful to write tests in such a way that the reader never has to leave the test method to understand what's going on. In this case, it looks like `getMessage()` _always_ returns something for `https://localhost:8829` and `mailboxId` (which would be problematic in production!) until seeing the wider class and that those values are configured above.

In this case I would propose a helper method (or more than one) that accepts only the configuration options we actually care about and returns the `MeshRequests`. That way, the reader never needs to leave the test method to see why we expect that exact string in the resultant request URI.

**Alternatively**
This is working code we're pulling from another codebase, so we might want to touch it as little as possible.

Reply:
I 100% agree with this, helper methods would no doubt make testing easier and remove the problematic hard coded values in multiple places. For now (as i don't want to break too many things and want to unblock the next tickets), i'll pull the duplicated code in this particular test class into constants for readability. Then we can refactor later - hope that's ok

Reply:
Sure thing!